rocAL/rocAL/source/augmentations/geometry_augmentations/node_crop_mirror_normalize.cpp

```diff
@@ -26,59 +26,106 @@ THE SOFTWARE.
 #include "exception.h"
 
 
-CropMirrorNormalizeNode::CropMirrorNormalizeNode(const std::vector<Image *> &inputs,
-                                                 const std::vector<Image *> &outputs) :
+CropMirrorNormalizeNode::CropMirrorNormalizeNode(const std::vector<rocalTensor *> &inputs,
+                                                 const std::vector<rocalTensor *> &outputs) :
         Node(inputs, outputs),
-        _mirror(MIRROR_RANGE[0], MIRROR_RANGE[1])
-{
+        _mirror(MIRROR_RANGE[0], MIRROR_RANGE[1]) {
         _crop_param = std::make_shared<RocalCropParam>(_batch_size);
 }
 
-void CropMirrorNormalizeNode::create_node()
-{
+void CropMirrorNormalizeNode::create_node() {
     if(_node)
         return;
 
     if(_crop_param->crop_h == 0 || _crop_param->crop_w == 0)
         THROW("Uninitialized destination dimension - Invalid Crop Sizes")
 
     _crop_param->create_array(_graph);
-    _mean_vx.resize(_batch_size);
-    _std_dev_vx.resize(_batch_size);
-    for (uint i=0; i < _batch_size; i++ ) {
-         _mean_vx[i] = _mean;
-         _std_dev_vx[i] = _std_dev;
+
+    std::vector<float> multiplier_vec, offset_vec;
+    int multiplier_offset_array_size = _batch_size * _inputs[0]->info().get_channels();
+    if(!_std_dev[0])
+        THROW("Standard deviation value cannot be 0");
+    multiplier_vec.resize(multiplier_offset_array_size, -(_mean[0] / _std_dev[0]));
+    offset_vec.resize(multiplier_offset_array_size, (1 / _std_dev[0]));
+    
+    if(_inputs[0]->info().get_channels() == 3) {
+        if(!(_std_dev[0] && _std_dev[1] && _std_dev[2]))
+            THROW("Standard deviation value cannot be 0");
+        offset_vec[0] = 1 / _std_dev[0];
+        offset_vec[1] = 1 / _std_dev[1];
+        offset_vec[2] = 1 / _std_dev[2];
+        multiplier_vec[0] = -(_mean[0] * offset_vec[0]);
+        multiplier_vec[1] = -(_mean[1] * offset_vec[1]);
+        multiplier_vec[2] = -(_mean[2] * offset_vec[2]);
+        for (uint i = 1, j = 3; i < _batch_size; i++, j += 3) {
+            multiplier_vec[j] = multiplier_vec[0];
+            multiplier_vec[j + 1] = multiplier_vec[1];
+            multiplier_vec[j + 2] = multiplier_vec[2];
+            offset_vec[j] = offset_vec[0];
+            offset_vec[j + 1] = offset_vec[1];
+            offset_vec[j + 2] = offset_vec[2];
+        }
     }
-    _mean_array = vxCreateArray(vxGetContext((vx_reference)_graph->get()), VX_TYPE_FLOAT32, _batch_size);
-    _std_dev_array = vxCreateArray(vxGetContext((vx_reference)_graph->get()), VX_TYPE_FLOAT32, _batch_size);
+    _multiplier_vx_array = vxCreateArray(vxGetContext((vx_reference)_graph->get()), VX_TYPE_FLOAT32, multiplier_offset_array_size);
```

Comment:
I thought we have discussed to switch all the parameters to use tensor and pinned memory